### ISSUE TITLE
[Feature Request]: Options to disable public IPv4 and/or IPv6 when creating nodes

### ISSUE THREAD

**Issue by lpellegr** [kube-hetzner/terraform-hcloud-kube-hetzner#843]
### Description

This would help reduce fees and more. See the discussion started here:

https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/837

**Comment by aleksasiriski** [comment]
There are a few limiting factors with Hetzner IPv6 only VMs, read my reply [here](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/837#discussioncomment-6187040)

**Comment by mysticaltech** [comment]
@aleksasiriski Yes, for sure, but I have a feeling it's worth trying, maybe it will work out of the box..

**Comment by aleksasiriski** [comment]
I'm interested in this, so will give it a go when I'm done with uni tests next week.

**Comment by mysticaltech** [comment]
@aleksasiriski I remember that time, may the wind be at your back!

**Comment by LuigiClemente** [comment]
Another benefit is that we can use Nebula to create a private overlay network that spans across multiple cloud providers or data centers, such as AWS and Hetzner. This allows you to replicate a similar network topology and configuration in Hetzner as you have in AWS, with some modifications. Here's how it works:

1. **Nebula Overlay Network**: Nebula creates a secure, private overlay network that operates independently of the underlying infrastructure. You can deploy Nebula nodes on servers in different cloud providers, including AWS and Hetzner.
    
2. **Replicating AWS Topology in Hetzner**:
    
    * **Servers**: You can set up servers in Hetzner that correspond to the instances you have in AWS. These servers become part of the Nebula overlay network.
    * **Security Groups**: While AWS uses security groups, Hetzner may use firewall rules. You'll need to adapt your security configurations to the specific cloud provider.
    * **VPC/Subnet**: In AWS, you have Virtual Private Cloud (VPC) and subnets. In Hetzner, you might have different network configurations. Nebula abstracts these differences by creating a private network between your nodes.
3. **Cross-Cloud Communication**: With Nebula, servers in Hetzner can securely communicate with servers in AWS over the Nebula overlay network. This network operates independently of the public internet, providing a secure and private communication channel.
    
4. **Terraform Modifications**: While you can use Terraform to manage infrastructure in both AWS and Hetzner, you'll need to make adjustments to your Terraform configurations to accommodate the differences in cloud providers. This includes changes to resource types, configurations, and security settings.
    
5. **Security and Isolation**: Nebula allows you to define network isolation and security policies, ensuring that communication between cloud providers is secure and controlled.
    

By using Nebula as a bridge between your cloud providers, you can create a unified private network that connects resources in different environments, such as AWS and Hetzner, with relative ease. However, keep in mind that there may still be some cloud-specific configurations and adjustments required in your Terraform code to match the nuances of each provider's infrastructure.
